nightfall-deployer/contracts/ChallengesUtil.sol

```diff
@@ -156,15 +156,19 @@ library ChallengesUtil {
         for (uint256 i = 0; i < transaction.proof.length; i++) {
             if (transaction.proof[i] == 0) nZeroProof++;
         }
+        uint256 nZeroCompressedSecrets;
+        for (uint256 i = 0; i < transaction.compressedSecrets.length; i++) {
+            if (transaction.compressedSecrets[i] == 0) nZeroCompressedSecrets++;
+        }
         require(
                 (transaction.tokenId == ZERO && transaction.value == 0) ||
                 transaction.ercAddress == ZERO ||
                 transaction.recipientAddress == ZERO ||
-                transaction.commitments.length != 0 ||
+                transaction.commitments[0] != ZERO ||
+                transaction.commitments[1] != ZERO ||
                 transaction.nullifiers[0] == ZERO ||
                 transaction.nullifiers[1] != ZERO ||
-                transaction.nullifiers.length != 1 ||
-                transaction.compressedSecrets.length != 0 ||
+                transaction.compressedSecrets.length != nZeroCompressedSecrets ||
```

Comment:
Actually I take my comment back. The exception case I suggested is wrong because the length of the array is always 8 and will never be `transaction.compressedSecrets.length == 4`

But can I request that we do the straight forward check of `nZeroCompressedSecrets != 8 ||` here please?
This logic you wrote is quite clever but it takes a long time to interpret the negations and anyone looking at might be as confused to until a much deeper look. Apologies for the ask Hari :)

Reply:
No worries . Changes done.